### ISSUE TITLE
restoreWindows setting ignored 

### ISSUE THREAD

**Issue by fonnesbeck** [Microsoft/vscode#42062]
This is an ongoing problem first described in #41039, which was not re-opened when I noticed the problem had continued, so I am opening a new issue.

- VSCode Version: Code 1.19.2 (490ef761b76b3f3b3832eff7a588aac891e5fe80, 2018-01-10T15:49:04.682Z)
- OS Version: Darwin x64 17.4.0
- Extensions: 

Extension|Author (truncated)|Version
---|---|---
latex-workshop|Jam|3.12.3
django-html|bib|1.2.0
django-snippets|bib|1.1.0
jupyter|don|1.1.4
python-extension-pack|don|1.0.1
file-icons|fil|1.0.7
shell-format|fox|1.1.1
mdmath|goe|2.1.0
Python-autopep8|him|1.0.2
nbpreviewer|jit|1.0.0
MagicPython|mag|1.0.12
python|ms-|0.9.1
jinja|who|0.0.8
markdown-pdf|yza|0.1.7
markdown-all-in-one|yzh|1.0.3

(6 theme extensions excluded)

---

Steps to Reproduce:

- Open files with vscode
- Set the following preference "window.restoreWindows": "none"
- Close vscode
- Restart vscode
- Observe previous documents open, despite setting
- When I open a new session under the setting above, I expect no open windows, yet previous windows always appear on startup.

I have tried using insider builds, but the problem still arises. 

Reproduces without extensions: Yes

**Comment by fonnesbeck** [comment]
I have also tried setting:

    "window.reopenFolders": "none"

but the issue continues

**Comment by bpasero** [comment]
@fonnesbeck are you starting code from the command line with "`code .`"?

**Comment by fonnesbeck** [comment]
It is. Can this be turned off?

**Comment by bpasero** [comment]
@fonnesbeck so what is the issue here then. You ask code to open a folder and it opens, what else would you expect?

**Comment by fonnesbeck** [comment]
Sorry for not being clear. I do expect the folder to open up, but what I did not expect is the 2 or 3 files that I happened to have open the last time (including some not from that folder) to also be opened. It appears to be restoring a session, and not just giving me access to the folder.

So for example, if I call `code .` in one of my repositories, I get this:

![](https://d.pr/i/mNfHhi+)

which shows the folder in the sidebar (I expect this), but also 4 files from the last session, which I am not interested in seeing. 

Whereas, if I run `code .` in a folder that I have not recently used vscode in, I get this:

![](https://d.pr/i/k7imuv+)

This is what I would expect when I set `"window.restoreWindows": "none"` -- no windows restored.
